src/main/java/hudson/plugins/git/browser/ViewGitWeb.java

```diff
@@ -89,29 +93,37 @@ public ViewGitWeb newInstance(StaplerRequest req, @Nonnull JSONObject jsonObject
         }
 
         @RequirePOST
-        public FormValidation doCheckUrl(@QueryParameter(fixEmpty = true) final String url) throws IOException, ServletException {
-            if (url == null) // nothing entered yet
-                return FormValidation.ok();
-            // Connect to URL and check content only if we have admin permission
-            if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER))
+        public FormValidation doCheckRepoUrl(@AncestorInPath Item project, @QueryParameter(fixEmpty = true) final String repoUrl)
+                throws IOException, ServletException, URISyntaxException {
+
+            String cleanUrl = Util.fixEmptyAndTrim(repoUrl);
+            if(initialChecksAndReturnOk(project, cleanUrl)){
                 return FormValidation.ok();
-            return new URLCheck() {
-                protected FormValidation check() throws IOException, ServletException {
-                    String v = url;
-                    if (!v.endsWith("/"))
-                        v += '/';
-
-                    try {
-                        if (findText(open(new URL(v)), "ViewGit")) {
-                            return FormValidation.ok();
-                        } else {
-                            return FormValidation.error("This is a valid URL but it doesn't look like ViewGit");
+            }
+            FormValidation response;
```

Comment:
I don't think this temporary is required, since the call to checkURIFormat can return from both the true and the false conditions.
```suggestion
```

Reply:
Agreed, changed it.